src/lib.rs

```diff
@@ -138,6 +118,15 @@ impl Beanstalkd {
         Beanstalkd { connection: bean }
     }
 
+    async fn response(&mut self) -> Result<AnyResponse, proto::error::Decode> {
+        use proto::error::{ErrorKind, ProtocolError};
+
+        match self.connection.next().await {
+            Some(r) => r,
+            None => Err(ErrorKind::Protocol(ProtocolError::StreamClosed).into()),
```

Comment:
This isn't really a protocol error but I'm unsure where it should belong. It's private to the crate so the decision doesn't need to be made now.